relay-server/src/http.rs

```diff
@@ -219,7 +215,7 @@ where
 {
     // Local imports because importing them top-level is too confusing with two futures crates
     use futures03::{stream::poll_fn, task::Poll};
-    let mut buf = Vec::with_capacity(8192);
+    let mut buf = vec![0; 8192];
```

Comment:
@jan-auer wdyt, stack-allocate this array and make it smaller, or keep in heap and let it stay big?

Reply:
+1 on stack, even with 8k. You‘ll have to move the buffer into the `poll_fn` closure, and with that it will be moved to the heap due to `move` semantics.